compiler/rustc_session/src/session.rs

```diff
@@ -1305,7 +1305,7 @@ pub fn build_session(
         early_error(sopts.error_format, &format!("Error loading host specification: {}", e))
     });
 
-    let loader = file_loader.unwrap_or(Box::new(RealFileLoader));
+    let loader = file_loader.unwrap_or_else(|| Box::new(RealFileLoader));
```

Comment:
As `Box::new` causes an allocation, I would prefer this change to use the closure.

Reply:
`RealFileLoader` is a ZST so `Box::new` doesn't actually allocate here. Using `unwrap_or_else` is still fine here though

Reply:
Ok, leaving `unwrap_or_else` b/c `Box` in a majority of cases could allocate and it would be better to hide that behind a closure.